implementation/core/src/main/java/io/smallrye/faulttolerance/core/apiimpl/LazyFaultTolerance.java

```diff
@@ -36,15 +37,13 @@ public void run(Runnable action) {
     }
 
     private FaultTolerance<T> instance() {
-        FaultTolerance<T> instance = this.instance;
+        FaultTolerance<T> instance = this.instance.get();
```

Comment:
this is making `instance::get` to be called more than once, changing the expected behaviour for users.
Using ReentrantLock(s) is preferrable here - or use a 2 state lock-free behaviour instead